### ISSUE TITLE
GKD with unsloth not working

### ISSUE THREAD

**Issue by deekshaVarshney** [huggingface/trl#3027]
This is the code which i am running

from unsloth import FastLanguageModel
from unsloth.chat_templates import get_chat_template
import torch
import re
from unsloth.chat_templates import standardize_sharegpt
from trl import SFTTrainer
from transformers import TrainingArguments, DataCollatorForSeq2Seq
from unsloth import is_bfloat16_supported
from unsloth.chat_templates import train_on_responses_only
from unsloth.chat_templates import get_chat_template
from transformers import TextStreamer
import pandas as pd
from datasets import Dataset
from sklearn.model_selection import train_test_split
from datasets import load_dataset, Dataset
from trl import GKDConfig, GKDTrainer
from transformers import AutoModelForCausalLM, AutoTokenizer, TrainingArguments, pipeline
import numpy as np
from evaluate import load  # Hugging Face's evaluate library

from peft import LoraConfig, get_peft_model, TaskType
from transformers import BitsAndBytesConfig

max_seq_length = 2048 # Choose any! We auto support RoPE Scaling internally!
dtype = None # None for auto detection. Float16 for Tesla T4, V100, Bfloat16 for Ampere+
load_in_4bit = True # Use 4bit quantization to reduce memory usage. Can be False.

student_model_path = "/home/deeksha/codes/Qwen2.5-0.5B-Instruct"
teacher_model_path = "/root/home/deeksha/codes/student/models/qwen_2.5_model_1.5B_fine"

train_data = pd.read_csv('/root/home/deeksha/codes/annotated_data/data/formatted_training_data.csv')
train_dataset = Dataset.from_pandas(train_data)

# Format dataset for training
def format_example(example):
    return {
        "messages": [
            {"role": "user", "content": example["prompt"]},
            {"role": "assistant", "content": example["completion"]},
        ]
    }

train_dataset = train_dataset.map(format_example, remove_columns=train_dataset.column_names)
train_dataset = Dataset.from_dict({"messages": train_dataset["messages"]})

# Optional: Check the structure of the dataset
print(train_dataset)

test_data = pd.read_csv('/root/home/deeksha/codes/annotated_data/data/formatted_testing_data.csv')
test_data = test_data.head(10)
test_dataset = Dataset.from_pandas(test_data)
test_dataset = test_dataset.map(format_example, remove_columns=test_dataset.column_names)

eval_dataset = test_dataset
print(test_dataset)

# Load the tokenizer
tokenizer = AutoTokenizer.from_pretrained(student_model_path)
teacher_model = AutoModelForCausalLM.from_pretrained(teacher_model_path)

student_model, tokenizer = FastLanguageModel.from_pretrained(
    model_name = student_model_path,        #"unsloth/Llama-3.2-3B-Instruct", # or choose "unsloth/Llama-3.2-1B-Instruct"
    max_seq_length = max_seq_length,
    dtype = dtype,
    load_in_4bit = load_in_4bit,
    # token = "hf_...", # use one if using gated models like meta-llama/Llama-2-7b-hf
)

training_args = GKDConfig(
    output_dir="models/gkd-model-unsloth",
    per_device_train_batch_size=2,
    per_device_eval_batch_size=2,
    evaluation_strategy="steps",
    eval_steps=10,
    save_steps=2,
    save_total_limit=3,
    logging_steps=100,
    learning_rate=5e-5,  # Can be increased (e.g., 1e-4 for LoRA)
    num_train_epochs=1,
    weight_decay=0.0,  # No weight decay for LoRA
    report_to="none",
    seq_kd=True,
)

# Initialize Trainer
trainer = GKDTrainer(
    model=student_model,  # LoRA-applied model
    teacher_model=teacher_model,
    args=training_args,
    processing_class=tokenizer,
    train_dataset=train_dataset,
    eval_dataset=test_dataset,
)

# **Train the model**
trainer.train()

**Comment by qgallouedec** [comment]
Hey @deekshaVarshney, there's no error message, traceback, or description of what "not working" means in your report, and the script references local CSVs we can't access, so there's no way to investigate from here. I'm going to close this out; happy to re-open if you can share the actual error/traceback, the output of `trl env`, and a minimal repro that doesn't depend on private data. (Heads up: GKD has since moved to `trl.experimental.gkd` on current main, so the import path in your script will also need updating.)
